src/core/model/Document.cpp

```diff
@@ -128,21 +132,52 @@ auto Document::createSaveFolder(fs::path lastSavePath) -> fs::path {
     return lastSavePath;
 }
 
-auto Document::createSaveFilename(DocumentType type, const std::string& defaultSaveName) -> fs::path {
+std::string Document::parseFilenameWildcard(const std::string& wildcard) {
+    if (wildcard == "name") {
+        fs::path pdfPath = this->pdfFilepath.filename();
+        Util::clearExtensions(pdfPath, ".pdf");
+        return pdfPath.u8string();
+    }
+    if (wildcard == "date" || wildcard == "time") {
+        std::time_t time = std::chrono::system_clock::to_time_t(
+            std::chrono::system_clock::now()
+        );
+        std::stringstream timestring;
+        timestring << std::put_time(std::localtime(&time),
+                                wildcard == "date" ? "%Y-%m-%d" : "%X");
+        return timestring.str();
+    } 
+    // not a valid wildcard
+    return "";
+}
+
+auto Document::createSaveFilename(DocumentType type, const std::string& defaultSaveName, const std::string& defaultPfdName) -> fs::path {
     if (!filepath.empty()) {
         // This can be any extension
         fs::path p = filepath.filename();
         Util::clearExtensions(p);
         return p;
     }
     if (!pdfFilepath.empty()) {
-        fs::path p = pdfFilepath.filename();
-        if (this->attachPdf) {
-            Util::clearExtensions(p, ".pdf");
-        } else {
-            Util::clearExtensions(p);
+        // Build the pdf-filepath according to settings (replace wildcards)
+        std::string saveString = defaultPfdName;
+        size_t pos = saveString.find(DEFAULT_PDF_WILDCARD_START);
+
+        while (pos != std::string::npos) {
+            size_t endPos = saveString.find(DEFAULT_PDF_WILDCARD_END, pos + 2);
```

Comment:
The constant `2` is the length of `DEFAULT_PDF_WILDCARD_START`:
```suggestion
            size_t endPos = saveString.find(DEFAULT_PDF_WILDCARD_END, pos + std::strlen(DEFAULT_PDF_WILDCARD_START));
```

Reply:
Oh yes, seems way better than this hard-coded stuff. Thank you :)